services/edit/index.js

```diff
@@ -292,16 +328,26 @@ function cloneChildComponents(res) {
  */
 function createComponent(name, data) {
   var base = site.get('prefix') + '/components/' + name,
-    instance = base + '/instances';
+    instance = base + '/instances',
+    getDefaultData;
 
   if (data) {
-    return cache.createThrough(instance, data);
+    // if we passed through data, use it
+    getDefaultData = Promise.resolve(data);
   } else {
-    return cache.getDataOnly(base) // create component with base JSON from bootstrap.
-      .then(function (baseJson) {
-        return cache.createThrough(instance, baseJson).then(cloneChildComponents);
-      });
+    // otherwise get the component's default data
+    getDefaultData = cache.getDataOnly(base);
   }
+
+  return getDefaultData.then(function (defaultData) {
+    let children = getChildComponents(defaultData);
```

Comment:
The _major_ philosophical difference with this change is that we're figuring out the child components and creating them **in parallel**, rather than waiting until we've already created the parent before looking for children.